blog/2023-02-20-introducing-the-dhis2-java-sdk.md

```diff
@@ -0,0 +1,174 @@
+---
+slug: 2023/02/introducing-the-dhis2-java-sdk
+title: Introducing the DHIS2 Java SDK
+authors: claudemamo
+tags: [dhis2-java-sdk, integration]
+---
+
+At HISP Centre, we've been engineering numerous run-of–the-mill integration projects where data is exchanged back and forth with DHIS2 through its [Web API](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html). Typically, this data is filtered, mapped, transformed, enriched, and routed to or from an application which could be the same DHIS2 instance, a different instance, or even a distinct application altogether. Our programming language of choice for these projects is usually Java. Java has an [excellent ecosystem](https://camel.apache.org/), but in each project, we kept reinventing the wheel in terms of shipping the data out of and into DHIS2. Our productivity was being disturbed with the donkey work of writing code to serialise in-memory objects, deserialise JSON, handle network failures, authenticate, log events, pool remote connections, retry HTTP errors, and so on. We needed a lightweight flexible library that implemented the low-level plumbing. To this end, the DHIS2 Java SDK was born.
+
+[DHIS2 Java SDK](https://github.com/dhis2/dhis2-java-sdk) is a small open-source non-opinionated library that aims to provide the nuts and bolts for consuming the DHIS2 Web API. Specifically, it provides a fluent-like interface for creating, fetching, modifying, and deleting DHIS2 resources. The application developer can focus on the domain logic and not get bogged down with low-level details concerning DHIS2 API communication like authentication, retries, and pagination. Furthermore, with its type-safe resource model, the SDK aids application developers in writing client code that is compatible with the version of DHIS2 they’re integrating with. 
```

Comment:
```suggestion
The [DHIS2 Java SDK](https://github.com/dhis2/dhis2-java-sdk) is a small, open-source, and non-opinionated library that is specifically designed to provide the necessary tools and components for consuming the DHIS2 Web API. 

Specifically, it provides a fluent-like interface for creating, retrieving, updating, and deleting (`CRUD`) DHIS2 resources. The application developer can focus on the domain logic and not get bogged down with low-level details concerning DHIS2 API communication like authentication, retries, and pagination. Furthermore, with its type-safe resource model, the DHIS2 Java SDK aids application developers in writing client code that is compatible with the version of DHIS2 they’re integrating with. 
```

Reply:
@claudemamo could you remove the "nuts and bolts" expression here and replace it with something like "necessary tools"? It's not an expression everyone will understand

Reply:
Same with "donkey work"